src/test/java/com/uid2/shared/store/RotatingServiceLinkStoreTest.java

```diff
@@ -62,36 +65,46 @@ public void loadContentEmptyArray() throws Exception {
     }
 
     @Test
-    public void loadContentMultipleServices() throws Exception {
+    public void loadContent_MultipleServiceLinksStored_LoadsAllServiceLinks() throws Exception {
         JsonArray content = new JsonArray();
-        ServiceLink l1 = addServiceLink(content, "abc123", 1, 123, "Test Service 1");
-        ServiceLink l2 = addServiceLink(content, "abc123", 2, 123, "test1");
-        ServiceLink l3 = addServiceLink(content, "ghi789", 1, 123, "Test Service 1");
-        ServiceLink l4 = addServiceLink(content, "jkl1011", 3, 124, "test2");
+        ServiceLink l1 = addServiceLink(content, "abc123", 1, 123, "Test Service 1", Set.of());
+        ServiceLink l2 = addServiceLink(content, "abc123", 2, 123, "test1", Set.of(Role.MAPPER));
+        ServiceLink l3 = addServiceLink(content, "ghi789", 1, 123, "Test Service 1", Set.of(Role.MAPPER, Role.SHARER));
+        ServiceLink l4 = addServiceLink(content, "jkl1011", 3, 124, "test2", null);
         when(cloudStorage.download("locationPath")).thenReturn(makeInputStream(content));
 
         final long count = serviceLinkStore.loadContent(makeMetadata("locationPath"));
         assertEquals(4, count);
         assertTrue(serviceLinkStore.getAllServiceLinks().containsAll(Arrays.asList(l1, l2, l3, l4)));
     }
+
     @Test
-    public void findServiceLinksMultipleServices() throws Exception {
+    public void getServiceLink_MultipleServiceLinksStored_FindsCorrectServiceLink() throws Exception {
         JsonArray content = new JsonArray();
-        ServiceLink l1 = addServiceLink(content, "abc123", 1, 123, "Test Service 1");
-        ServiceLink l2 = addServiceLink(content, "abc123", 2, 123, "test1");
-        ServiceLink l3 = addServiceLink(content, "ghi789", 1, 123, "Test Service 1");
-        ServiceLink l4 = addServiceLink(content, "jkl1011", 3, 124, "test2");
+        ServiceLink l1 = addServiceLink(content, "abc123", 1, 123, "Test Service 1", Set.of());
+        ServiceLink l2 = addServiceLink(content, "abc123", 2, 123, "test1", Set.of(Role.MAPPER));
+        ServiceLink l3 = addServiceLink(content, "ghi789", 1, 123, "Test Service 1", Set.of(Role.MAPPER, Role.SHARER));
+
         when(cloudStorage.download("locationPath")).thenReturn(makeInputStream(content));
 
         final long count = serviceLinkStore.loadContent(makeMetadata("locationPath"));
 
-        ServiceLink sl = serviceLinkStore.getServiceLink(1, "abc123");
-        assertNotNull(sl);
-        assertEquals("Test Service 1", sl.getName());
-        assertEquals(1, sl.getServiceId());
-        assertEquals(123, sl.getSiteId());
-        assertEquals("abc123", sl.getLinkId());
+        assertEquals(Arrays.asList(serviceLinkStore.getServiceLink(1, "abc123"),
```

Comment:
I think just to make it a bit clearer, this can be changed to:
```
List<ServiceLink> expected = List.of(l1, l2, l3);
List<ServiceLink> actual = List.of(
  serviceLinkStore.getServiceLink(1, "abc123"),
  serviceLinkStore.getServiceLink(2, "abc123"),
  serviceLinkStore.getServiceLink(1, "ghi789")
);
assertEquals(expected, actual);
```